data-prepper-plugins/parquet-codecs/src/test/java/org/opensearch/dataprepper/plugins/parquetinputcodec/parquetInputCodecTest.java

```diff
@@ -0,0 +1,173 @@
+
+    private ParquetInputCodec createObjectUnderTest() {
+        return new ParquetInputCodec();
+    }
+
+    @BeforeEach
+    void setup() {
+        parquetInputCodec = createObjectUnderTest();
+    }
+
+    @Test
+    void test_when_nullInputStream_then_throwsException() {
+        assertThrows(NullPointerException.class, () ->
+                parquetInputCodec.parse(null, eventConsumer));
+
+        verifyNoInteractions(eventConsumer);
+    }
+
+    @Test
+    void parse_with_null_Consumer_throws() throws IOException {
+        parquetInputCodec = createObjectUnderTest();
+
+        final InputStream inputStream = mock(InputStream.class);
+        assertThrows(NullPointerException.class, () ->
+                parquetInputCodec.parse(inputStream, null));
+        verifyNoInteractions(inputStream);
+    }
+
+    @Test
+    void parse_with_empty_InputStream_does_not_call_Consumer() throws IOException {
+        final ByteArrayInputStream emptyInputStream = new ByteArrayInputStream(new byte[]{});
+        createObjectUnderTest().parse(emptyInputStream, eventConsumer);
+        verifyNoInteractions(eventConsumer);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {10})
+    void test_when_HappyCaseParquetInputStream_then_callsConsumerWithParsedEvents(final int numberOfRecords) throws IOException {
+        InputStream inputStream = createRandomParquetStream(numberOfRecords);
+
+        parquetInputCodec.parse(inputStream,eventConsumer);
+
+        final ArgumentCaptor<Record<Event>> recordArgumentCaptor = ArgumentCaptor.forClass(Record.class);
+        verify(eventConsumer, times(numberOfRecords)).accept(recordArgumentCaptor.capture());
+        final List<Record<Event>> actualRecords = recordArgumentCaptor.getAllValues();
+        assertThat(actualRecords.size(), equalTo(numberOfRecords));
+
+        for (int i = 0; i < actualRecords.size(); i++) {
+
+            final Record<Event> actualRecord = actualRecords.get(i);
+            assertThat(actualRecord, notNullValue());
+            assertThat(actualRecord.getData(), notNullValue());
+            assertThat(actualRecord.getData().getMetadata(), notNullValue());
+            assertThat(actualRecord.getData().getMetadata().getEventType(), equalTo(EventType.LOG.toString()));
+        }
+
+    }
+
+    private static InputStream createRandomParquetStream(int numberOfRecords) throws IOException {
+
+        Files.deleteIfExists(java.nio.file.Path.of("./test-parquet.parquet"));
+
+        Schema schema = parseSchema();
+        String OS = System.getProperty("os.name").toLowerCase();
+
+        if (OS.contains("win")) {
+            System.setProperty("hadoop.home.dir", Paths.get("").toAbsolutePath().toString());
+        } else {
+            System.setProperty("hadoop.home.dir", "/");
+        }
+
+        List<GenericData.Record> recordList = generateRecords(schema, numberOfRecords);
+        java.nio.file.Path path = Paths.get("test-parquet.parquet");
```

Comment:
Please use JUnit's `@TempDir` feature: https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory